File: app/src/main/java/forpdateam/ru/forpda/presentation/reputation/ReputationView.java
package forpdateam.ru.forpda.presentation.reputation;

import forpdateam.ru.forpda.api.reputation.models.RepData;
import forpdateam.ru.forpda.api.reputation.models.RepItem;
import forpdateam.ru.forpda.common.mvp.IBaseView;

/**
 * Created by radiationx on 03.01.18.
 */

public interface ReputationView extends IBaseView {

    void showReputation(RepData repData);

    void onChangeReputation(String result);

    void showItemDialogMenu(RepItem item);
}

File: app/src/main/java/forpdateam/ru/forpda/presentation/reputation/ReputationView.kt
package forpdateam.ru.forpda.presentation.reputation

import forpdateam.ru.forpda.common.mvp.IBaseView
import forpdateam.ru.forpda.entity.remote.reputation.RepData
import forpdateam.ru.forpda.entity.remote.reputation.RepItem
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.SkipStrategy
import moxy.viewstate.strategy.StateStrategyType

/**
 * Created by radiationx on 03.01.18.
 */

@StateStrategyType(AddToEndSingleStrategy::class)
interface ReputationView : IBaseView {
    fun showReputation(repData: RepData)

    @StateStrategyType(SkipStrategy::class)
    fun onChangeReputation()

    @StateStrategyType(SkipStrategy::class)
    fun showItemDialogMenu(item: RepItem)

    fun showAvatar(avatarUrl: String)
}
